src/config/ts-node.ts

```diff
@@ -15,42 +15,45 @@ const debug = Debug('ts-node')
 export const TS_CONFIGS: Record<string, TSConfig> = {}
 const REGISTERED = new Set<string>()
 
-function loadTSConfig(root: string): TSConfig | undefined {
-  if (TS_CONFIGS[root]) return TS_CONFIGS[root]
-  const tsconfigPath = join(root, 'tsconfig.json')
-  let typescript: typeof import('typescript') | undefined
+async function loadTSConfig(root: string): Promise<TSConfig | undefined> {
   try {
-    typescript = require('typescript')
-  } catch {
-    try {
-      typescript = require(require.resolve('typescript', {paths: [root, __dirname]}))
-    } catch {
-      debug(`Could not find typescript dependency. Skipping ts-node registration for ${root}.`)
-      memoizedWarn(
-        'Could not find typescript. Please ensure that typescript is a devDependency. Falling back to compiled source.',
-      )
-      return
-    }
-  }
+    if (TS_CONFIGS[root]) return TS_CONFIGS[root]
+    const tsconfigPath = join(root, 'tsconfig.json')
+    const tsconfig = await readJson<TSConfig>(tsconfigPath)
 
-  if (existsSync(tsconfigPath) && typescript) {
-    const tsconfig = typescript.parseConfigFileTextToJson(tsconfigPath, readJsonSync(tsconfigPath, false)).config
-    if (!tsconfig || !tsconfig.compilerOptions) {
-      throw new Error(
-        `Could not read and parse tsconfig.json at ${tsconfigPath}, or it ` +
-          'did not contain a "compilerOptions" section.',
-      )
-    }
+    if (!tsconfig || Object.keys(tsconfig.compilerOptions).length === 0) return
 
     TS_CONFIGS[root] = tsconfig
-    return tsconfig
+
+    if (tsconfig.extends) {
+      const {parse} = await import('tsconfck')
+      const result = await parse(tsconfigPath, {})
+      let tsNodeOpts = {}
+      for (const extended of (result.extended ?? []).reverse()) {
+        if (extended.tsconfig['ts-node']) {
+          tsNodeOpts = {...tsNodeOpts, ...extended.tsconfig['ts-node']}
+        }
+      }
```

Comment:
not 100% sure this works, but fromEntries() is LWW so you don't have to reverse the array to get the last ones in first for Object.assign.

```ts
const tsNodeOpts = Object.fromEntries(
        (result.extended ?? [])
          .map((e) => Object.entries(e.tsconfig['ts-node'] ?? {}))
      )
```